annotation-processor/src/main/java/org/hibernate/validator/ap/checks/ConstraintCheckFactory.java

```diff
@@ -189,7 +201,9 @@ public ConstraintCheckFactory(Types typeUtils, ConstraintHelper constraintHelper
 						new TargetCheck( annotationApiHelper ),
 						new ConstraintValidatorCheck( constraintHelper, annotationApiHelper ),
 						new AnnotationTypeMemberCheck( annotationApiHelper, typeUtils ),
-						new CrossParameterConstraintCheck( annotationApiHelper, constraintHelper, typeUtils )
+						new CrossParameterConstraintCheck( annotationApiHelper, constraintHelper, typeUtils ),
+						new AnnotationUserMessageCheck( annotationApiHelper ),
+						new AnnotationDefaultMessageCheck( annotationApiHelper, elementUtils )
```

Comment:
It was supposed to work for a case that I've just added with the new commit - `ValidMessageParameters.Case3`. But I only thought that this would work for it and didn't actually write the test case. So sorry for that. Now with the test case added I removed `AnnotationUserMessageCheck` from list of `CONSTRAINT_META_ANNOTATION` and added it to CONSTRAINT_ANNOTATION (in `annotationTypeChecks` map)